prow/plugins/trigger/pull-request.go

```diff
@@ -141,10 +149,63 @@ func handlePR(c Client, trigger plugins.Trigger, pr github.PullRequestEvent) err
 			}
 			return buildAll(c, &pr.PullRequest, pr.GUID, *trigger.ElideSkippedContexts, baseSHA, presubmits)
 		}
+	case github.PullRequestActionClosed:
+		if err := abortAllJobs(c, &pr.PullRequest); err != nil {
+			c.Logger.WithError(err).Error("Failed to abort jobs for closed pull request")
+			return err
+		}
 	}
 	return nil
 }
 
+func abortAllJobs(c Client, pr *github.PullRequest) error {
+	selector, err := labelSelectorForPR(pr)
+	if err != nil {
+		return fmt.Errorf("failed to construct label selector: %w", err)
+	}
+
+	listOpts := metav1.ListOptions{LabelSelector: selector.String()}
+	jobs, err := c.ProwJobClient.List(listOpts)
+	if err != nil {
+		return fmt.Errorf("failed to list prowjobs for pr: %w", err)
+	}
+
+	var errs []error
+	for _, job := range jobs.Items {
+		// Do not abort jobs that already completed
+		if job.Complete() {
+			continue
+		}
+		job.Status.State = prowapi.AbortedState
+		if _, err := c.ProwJobClient.Update(&job); err != nil {
+			errs = append(errs, fmt.Errorf("failed to abort job %s: %w", job.Name, err))
+		}
+	}
+
+	return utilerrors.NewAggregate(errs)
+}
+
+func labelSelectorForPR(pr *github.PullRequest) (klabels.Selector, error) {
```

Comment:
Wasn't there a utility to make a selector from a map, and in that case couldn't we make one from

```go
requirements := map[string]string{
    kube.OrgLabel: pr.Base.Repo.Owner.Login,
// ...
}
```
?

Reply:
or if not, 

```go
selector := klabels.NewSelector()
for key, value := range map[string]string{
    kube.OrgLabel: pr.Base.Repo.Owner.Login,
// ...
} {
    req, err := klabels.NewRequirement(key, selection.Equals, []string{value})
// ...
    selector.Add(*req)
}
return selector
```